metricflow/plan_conversion/dataflow_to_sql.py

```diff
@@ -1348,3 +1349,36 @@ def visit_join_to_time_spine_node(self, node: JoinToTimeSpineNode) -> SqlDataSet
                 where=where,
             ),
         )
+
+    def visit_min_max_node(self, node: MinMaxNode) -> SqlDataSet:  # noqa: D
+        parent_data_set = node.parent_node.accept(self)
+        parent_alias = self._next_unique_table_alias()
+
+        assert (
+            len(parent_data_set.sql_select_node.select_columns) == 1
+        ), "MinMaxNode supports exactly one parent select column."
+        parent_column_alias_expr = SqlStringExpression(parent_data_set.sql_select_node.select_columns[0].column_alias)
+
+        # Build aggregate columns, labeled "min" & "max"
+        select_columns = [
+            SqlSelectColumn(
+                expr=SqlAggregateFunctionExpression(
+                    sql_function=SqlFunction.from_aggregation_type(aggregation_type=agg_type),
+                    sql_function_args=[parent_column_alias_expr],
+                ),
+                column_alias=agg_type.value,
+            )
+            for agg_type in (AggregationType.MIN, AggregationType.MAX)
+        ]
+        return SqlDataSet(
+            instance_set=parent_data_set.instance_set,
```

Comment:
We haven't fleshed out the `MetadataSpec` too much, but the min / max value might be better associated with that.

Reply:
Are you thinking - instead of adding a `MinMaxNode`, use the `MetadataSpec` and add a new `InstanceSpecVisitor` to visit it?

Reply:
Oh please don't add more InstanceSpecVisitors.....

Reply:
FWIW, this would be right in line with the documented purpose of the MetadataSpec, although it's a bit fuzzier since the column is actually part of the output.

It would be nice if the contract around instance spec and SqlSelectStatementNode.select_columns was more clear and easier to inspect, but it's complicated.